### ISSUE TITLE
clang-tidy misc-use-internal-linkage false positive for function declaration without body

### ISSUE THREAD

**Issue by HerrCai0907** [llvm/llvm-project#117488]
`misc-use-internal-linkage` should only warn for the function declaration with body.

It is a common usage in C++ to declare external function at the beginning.

```c++
void f();

void bar() {
  f();
}
```
